src/learn/PaywallModal.js

```diff
@@ -17,8 +17,10 @@ function LoadingModal(props) {
 
   useEffect(()=>{
     let handleModalClose = (event) =>{
-      if(!modalRef.current?.contains(event.target)){
-        history.push('/'); 
+      if(showPayment && !modalRef.current?.contains(event.target)){
```

Comment:
Maybe this is why it stopped working? To clarify, it needs to redirect you back to home if regardless if you are on `showPayment` render or not. Just, if the paywall modal is up period - you should be able to leave.